acceleration-everest-03-dotnet/UnitTests/AppServices/CustomerAppServiceTests.cs

```diff
@@ -0,0 +1,131 @@
+using AutoMapper;
+using DomainModels.Models;
+using DomainServices.Interfaces;
+using FluentAssertions;
+using Moq;
+using System.Collections.Generic;
+using UnitTests.Fixtures.Customers;
+
+namespace UnitTests.AppServices;
+
+public class CustomerAppServiceTests
+{
+    private readonly Mock<ICustomerBankInfoAppService> _mockCustomerBankInfoAppService;
+    private readonly Mock<ICustomerBankInfoService> _mockCustomerBankInfoService;
+    private readonly CustomerBankInfoAppService _customerBankInfoAppService;
+    private readonly Mock<ICustomerService> _mockCustomerService;
+    private readonly CustomerAppService _customerAppService;
+    private readonly IMapper _mapper;
+
+    public CustomerAppServiceTests()
+    {
+        var config = new MapperConfiguration(cfg =>
+        {
+            cfg.CreateMap<Customer, CustomerResult>();
+            cfg.CreateMap<CreateCustomer, Customer>();
+            cfg.CreateMap<UpdateCustomer, Customer>();
+        });
+        _mapper = config.CreateMapper();
+        _mockCustomerService = new Mock<ICustomerService>();
+        _mockCustomerBankInfoService = new Mock<ICustomerBankInfoService>();
+        _mockCustomerBankInfoAppService = new Mock<ICustomerBankInfoAppService>();
+        _customerBankInfoAppService = new CustomerBankInfoAppService(_mockCustomerBankInfoService.Object, _mapper);
+        _customerAppService = new CustomerAppService(_mockCustomerBankInfoAppService.Object, _mockCustomerService.Object, _mapper);
+    }
+
+    [Fact]
+    public async void Should_CreateCustomer_Successfully()
+    {
+        // Arrange  
+        var createCustomerTest = CreateCustomerFixture.GenerateCreateCustomerFixture();
+        var customerTest = CustomerFixture.GenerateCustomerFixture();
+
+        _mockCustomerService.Setup(customerService => customerService.CreateAsync(It.IsAny<Customer>())).ReturnsAsync(It.IsAny<long>());
+        _mockCustomerBankInfoService.Setup(customerBankInfoService => customerBankInfoService.Create(It.IsAny<long>()));
+
+        // Action
+        var result = await _customerAppService.CreateAsync(createCustomerTest).ConfigureAwait(false);
+        _customerBankInfoAppService.Create(result);
+
+        // Assert
+        result.Should().NotBe(null);
+
+        _mockCustomerService.Verify(customerService => customerService.CreateAsync(It.IsAny<Customer>()), Times.Once);
+        _mockCustomerBankInfoService.Verify(customerBankInfoService => customerBankInfoService.Create(It.IsAny<long>()), Times.Once);
+    }
+
+    [Fact]
+    public async void Should_DeleteCustomer_Successfully()
+    {
+        // Arrange  
+        long idTest = 1;
+
+        _mockCustomerService.Setup(customerService => customerService.DeleteAsync(It.IsAny<long>()));
+        _mockCustomerBankInfoService.Setup(customerBankInfoService => customerBankInfoService.DeleteAsync(It.IsAny<long>()));
+
+        // Action
+        await _customerAppService.DeleteAsync(idTest).ConfigureAwait(false);
```

Comment:
```suggestion
        await _customerAppService.DeleteAsync(It.IsAny<long>()).ConfigureAwait(false);
```
Use e abuse dessa ferramenta quando o campo não é tão relevante

Reply:
Mesmo na Action? Achei que ali precisava ser utilizado um dado do Arrange.